src/couch/src/couch_file.erl

```diff
@@ -264,6 +265,19 @@ rename_file(Original) ->
         Else -> Else
     end.
 
+rename_dir(RootDelDir, Original, DbName) ->
+    DbDir = binary_to_list(DbName) ++ "_design",
+    [DbPureName | _R] = lists:reverse(filename:split(binary_to_list(DbName))),
```

Comment:
Where was the discussion on grouping them like that? I don't see any immediate reason why the first approach of just mirroring the shards/$range/dbname/_design... hierarchy under the .recovery directory.

And the particular case I was referring to is if your dbname has an embedded / in it which is feasible. Because given your example it reads to me like if we had a dbname <<"foo/bar">> then we'd end up with files under .recovery/bar/shards/... which would be wrong as far as I can tell.